api.go

```diff
@@ -0,0 +1,469 @@
+package anthropic
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+)
+
+// DefaultApiToken is set to the environment variable ANTHROPIC_API_KEY, if
+// it exists. It can be overridden by setting it directly. It is used as the
+// default API token for all conversations.
+var DefaultApiToken = ""
+
+// DefaultSettings is the default settings for the Conversation. It is used
+// as the default settings for all Conversations, and can be overridden by
+// setting it directly.
+var DefaultSettings = SampleSettings{
+	Model:       "claude-3-5-sonnet-20240620",
+	Version:     "2023-06-01",
+	Beta:        "", // "max-tokens-3-5-sonnet-2024-07-15"
+	MaxTokens:   4096,
+	Temperature: 0.0,
+}
+
+// Anthropic Claude API Messages
+// Each input message content may be either a single string or an array of
+// content blocks, where each block has a specific type. Using a string
+// for content is shorthand for an array of one content block of type "text".
+//
+// The following input messages are equivalent:
+//   {"role": "user", "content": "Hello, Claude"}
+//   {"role": "user", "content": [{"type": "text", "text": "Hello, Claude"}]}
+//
+// We strictly use the content block format for text, as it simplifies the
+// serialization and deserialization of messages.
+
+// Message is a single message in a conversation.
+type Message struct {
+	// Role of the message sender.
+	// Possible values are:
+	//   "user": The message is from the user.
+	//   "assistant": The message is from the assistant.
+	Role string `json:"role"`
+	// Content is the content of the message, and may be a single string or
+	// image, or an array of content blocks.
+	Content *[]ContentBlock `json:"content"`
```

Comment:
Mostly just curious, does this slice need to be a pointer to a slice? Given that slices themselves are already pointers